src/systems/nav.js

```diff
@@ -17,5 +35,17 @@ AFRAME.registerSystem("nav", {
     geometry.applyMatrix(mesh.matrixWorld);
     this.pathfinder.setZoneData(zone, Pathfinding.createZone(geometry));
     this.el.sceneEl.emit("nav-mesh-loaded");
+
+    if (this.environmentScene) {
```

Comment:
I can't imagine `this.environmentScene` here not being set, but guard against it because it set in a promise chain after `waitForDOMContentLoaded`. Should I assume this already exists, or maybe throw an error / warning if it doesn't? Or maybe write a flag like `navMeshLoadedEarly`? I would not expect that flag to ever get set, but am not sure when exactly the DOMContentLoaded callback/chained promise will execute. Advice?